src/BSON/Javascript.c

```diff
@@ -215,6 +216,41 @@ PHP_METHOD(Javascript, getScope)
 }
 /* }}} */
 
+/* {{{ proto array Javascript::jsonSerialize()
+*/
+PHP_METHOD(Javascript, jsonSerialize)
+{
+	php_phongo_javascript_t *intern;
+	bool has_scope;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	intern = Z_JAVASCRIPT_OBJ_P(getThis());
+
+	has_scope = (intern->scope && intern->scope->len);
+	array_init_size(return_value, has_scope ? 2 : 1);
+	ADD_ASSOC_STRINGL(return_value, "$code", intern->code, intern->code_len);
```

Comment:
This implementation is consistent with the current Extended JSON spec. As noted in the tests, libbson currently converts Javascript types to JSON by returning its code field as a string value (see: [CDRIVER-1335](https://jira.mongodb.org/browse/CDRIVER-1335)). It also does not properly parse `$code` and `$scope` fields when converting JSON to BSON, which causes the JSON we produce here to be read by libbson as a normal embedded document.

The [HHVM implementation](https://github.com/mongodb/mongo-hhvm-driver/pull/148) currently mimics libbson (see: [HHVM-156](https://jira.mongodb.org/browse/HHVM-158)); however, I think we should change its behavior to comply with the spec rather than libbson, which should itself be fixed in 1.6.0.